pkg/prometheus/promcfg.go

```diff
@@ -1405,9 +1408,13 @@ func getLimit(user uint64, enforced *uint64) uint64 {
 	return user
 }
 
-func generateAddressShardingRelabelingRules(relabelings []yaml.MapSlice, shards int32) []yaml.MapSlice {
+func generateAddressShardingRelabelingRules(relabelings []yaml.MapSlice, shards int32, isProbe bool) []yaml.MapSlice {
```

Comment:
It might be easier to use different function names

```suggestion
func generateAddressShardingRelabelingRules(relabelings []yaml.MapSlice, shards int32) []yaml.MapSlice {
   return generateAddressShardingRelabelingRulesWithSourceLabel(relabelings, shards, "__address__")
}

func generateAddressShardingRelabelingRulesForProbes(relabelings []yaml.MapSlice, shards int32) []yaml.MapSlice {
   return generateAddressShardingRelabelingRulesWithSourceLabel(relabelings, shards, "__param_target")
}

func generateAddressShardingRelabelingRulesWithSourceLabel(relabelings []yaml.MapSlice, shards int32, shardLabel string) []yaml.MapSlice {
```